modules/block-interactions/src/main/java/unnamed/mmo/blocks/BlockInteractionUtils.java

```diff
@@ -0,0 +1,48 @@
+package unnamed.mmo.blocks;
+
+import net.minestom.server.instance.block.Block;
+import net.minestom.server.item.Material;
+import net.minestom.server.tag.Tag;
+import net.minestom.server.utils.NamespaceID;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import unnamed.mmo.blocks.handlers.CropBlockData;
+
+import java.util.Objects;
+
+public class BlockInteractionUtils {
+    private static final String DOMAIN_NAME = "unnammedmmo";
+
+    public static final NamespaceID TILL_HANDLER_ID = NamespaceID.from(DOMAIN_NAME, "tillhandler");
+    public static final NamespaceID FARMLAND_HANDLER_ID = NamespaceID.from(DOMAIN_NAME, "farmlandhandler");
+    public static final NamespaceID CROP_HANDLER_ID = NamespaceID.from(DOMAIN_NAME, "crophandler");
+
+
+    public static @NotNull Block storeDataOntoBlock(@NotNull Block block, @NotNull CropBlockData data) {
+         return block
```

Comment:
You should look into `Tag.Structure`, you can make a tag from a record and use that directly to clean this up slightly

Reply:
The standard record serializer doesn't support serializing Material and Block, so I quickly wrote my own serializer for it.